mesonbuild/interpreter/dependencyfallbacks.py

```diff
@@ -361,7 +361,7 @@ def lookup(self, kwargs: TYPE_nkwargs, force_fallback: bool = False) -> Dependen
                     identifier = dependencies.get_dep_identifier(name, kwargs)
```

Comment:
Each subproject has it's own current location, so If you have A => B => C, A will be paused on a call to `dependency()` or `subproject()` while it waits for B, which will be doing the same for C. By putting it in the local state we don't have to restore any state for B when C completes, and if C fails we don't have to worry about it, because B and A still have proper state.

Reply:
Thanks for the explanation. This has been resolved